make/Makefile.native

```diff
@@ -69,11 +70,20 @@ test_bin: $(TEST_BIN)
 $(TEST_BIN) : % : %.o $(CATCH_OBJ) $(BUILD_SRCS_OBJS) $(BUILD_TESTS_COMMON)
 	$(CXX) $(CXXFLAGS) $(INCLUDES) $^ -o $@ $(LFLAGS)
 
-run-tests: $(TEST_BIN) $(BUILD_PY_TESTS)
+run-tests:
 	$(MAKE) -f make/Makefile.native --no-print-directory tests
 	$(MAKE) -f make/Makefile.native --no-print-directory exec
-	@$(foreach t, $^, $(eval RET_VALS += $(shell echo $(t) 1>&2; ./$(t) 1>&2; echo $$?)))
-	@$(if $(filter-out 0, $(RET_VALS)), exit 1)
+	@HAS_ERROR=0;				\
+	for t in $(ALL_TESTS); do 		\
+		echo $$t; ./$$t;		\
```

Comment:
This looks much better :) One thing I might add is to pipe stdout/stderr to a temporary file and echo it only when an error occurs. That way it is easier to identify the failing tests.